plutus-core/plutus-core/src/PlutusCore/Evaluation/Machine/CostModelInterface.hs

```diff
@@ -88,9 +90,24 @@ extractParams cm = case toJSON cm of
     _ -> Nothing
 
 
+data CostModelApplyError =
+      CMUnknownParamError Text.Text
+    | CMReadInputError
+    | CMGenOutputError
+instance Exception CostModelApplyError
```

Comment:
do we want this?

Reply:
I am using it somewhere for convenience , so why not.